### PR TITLE
OHRM5X-1090: MyAttendace record GETALL API added

### PR DESCRIPTION
**Pull request by VageeshOrangeHRM**
This PR Contains following changes:
1. OHRM5X-1090 : Attendance-View Records: [API] Develop my attendance record getAll API added.
2. OHRM5X-1089:  Attendance-View Records: [API] Develop employee attendance record getAll added (Break Down).
3. Data group permissions and role permission added for the endpoint.
4. Unit test and Integration test not added.

### CODE REVIEW

File: symfony/plugins/orangehrmAttendancePlugin/Api/EmployeeAttendanceRecordAPI.php [hunk 1]
```diff
@@ -63,15 +69,77 @@ class EmployeeAttendanceRecordAPI extends Endpoint implements CrudEndpoint
      */
     public function getAll(): EndpointResult
     {
```
Comment (VageeshOrangeHRM) [inline symfony/plugins/orangehrmAttendancePlugin/Api/EmployeeAttendanceRecordAPI.php:71]: This is needed to be finalized. But for code reusing purpose i added this fragment and this is for employee attendance record view break  down list.

File: symfony/plugins/orangehrmAttendancePlugin/config/routes.yaml [hunk 2]
```diff
@@ -45,6 +45,13 @@ apiv2_attendance_employee_attendance_records:
     _key: empNumber
     _api: OrangeHRM\Attendance\Api\EmployeeAttendanceRecordAPI
 
+apiv2_attendance_employee_attendance_recordss:
+  path: /api/v2/attendance/employee/records
+  controller: OrangeHRM\Core\Controller\Rest\V2\GenericRestController::handle
+  methods: [ GET ]
+  defaults:
+    _api: OrangeHRM\Attendance\Api\EmployeeAttendanceRecordAPI
+
```
Comment (VageeshOrangeHRM) [inline symfony/plugins/orangehrmAttendancePlugin/config/routes.yaml:54]: Need to discuss with the naming

File: symfony/plugins/orangehrmAttendancePlugin/config/routes.yaml [hunk 3]
```diff
@@ -45,6 +45,13 @@ apiv2_attendance_employee_attendance_records:
     _key: empNumber
     _api: OrangeHRM\Attendance\Api\EmployeeAttendanceRecordAPI
 
+apiv2_attendance_employee_attendance_record_breakdown:
+  path: /api/v2/attendance/employee/records
+  controller: OrangeHRM\Core\Controller\Rest\V2\GenericRestController::handle
+  methods: [ GET ]
+  defaults:
+    _api: OrangeHRM\Attendance\Api\EmployeeAttendanceRecordAPI
```
Comment (VageeshOrangeHRM) [inline symfony/plugins/orangehrmAttendancePlugin/config/routes.yaml:53]: Need to discuss the naming convention?

**Review by RajithaKumara — APPROVED** [review]
Will add integrating test along with employee attendance record getAll operation development

**Pull request merged**